### ISSUE TITLE
Check that `length` assay exists when processing `tximeta` SE

### ISSUE THREAD

**Issue by csoneson** [mikelove/DESeq2#38]
When using a `tximeta` SE as the input to `DESeqDataSet`, even if the `length` assay has been manually removed, this code still tells the user that it's "using counts and average transcript lengths from tximeta" (and the check that all the values in the missing assay are >0 is not raising an error): https://github.com/mikelove/DESeq2/blob/1fce7c80e08d0231b0897580cec9fd9e3ead199c/R/AllClasses.R#L497-L504

While the rest of the downstream analysis goes through without errors, the message might be confusing. I guess there are two ways around this, not sure which one would be preferable:
* if the `length` assay does not exist, use just the counts (i.e., add a check to the `if` statement)
* if `length` should be there (based on the `countsFromAbundance`), stop if it is not available

**Comment by mikelove** [comment]
Thanks Charlotte, I've just added a check in 5a5a8f1 which should produce this message:

```
> dds <- DESeqDataSet(gse, design=~condition)                                                                                                                
 length assay is missing, using original counts without length adjustment                                                                                     
 ```

This looks scary enough that I think people will realize they are doing something not recommended.

**Comment by csoneson** [comment]
Yep, this looks great. Tried it on my example and I do indeed get that message :) Thanks!

**Comment by mikelove** [comment]
Thanks for catching this!

I was just checking and tximeta went over 2,000 downloads in March from ~1,200 addresses. That's pretty cool because tximport had 4,800 so we are nearly the 50% mark of tximport users.

**Comment by csoneson** [comment]
That's awesome 🚀😃
